### PR TITLE
Feat/result component

### PR DESCRIPTION
**Pull request by sendya**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> Add react `Result` component.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> Allow edits from maintainers.

### CODE REVIEW

**Comment by tangjinzhou** [comment]
是基于 antd 哪个分支去做的 统一基于antd 3.23.6的tag分支同步吧

**Comment by sendya** [comment]
> 是基于 antd 哪个分支去做的 统一基于antd 3.23.6的tag分支同步吧

我是基于 2019/10/01 的最后一个提交 https://github.com/ant-design/ant-design/commit/0a09b3b0085dc2715f6725602bcd04eacf1edb25 同步的我跟踪了一下，到 3.23.6 tag 相差 4 个 commit，没有对 Result 有进行修改的 commit。

只有 `Result` 有一个 `jest` 的快照结果更新

File: components/carousel/index.jsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ export const CarouselProps = {
   variableWidth: PropTypes.bool,
   useCSS: PropTypes.bool,
   slickGoTo: PropTypes.number,
+  responsive: PropTypes.array.def(null),
```
Comment (tangjinzhou) [inline components/carousel/index.jsx]: ？

File: components/tooltip/Tooltip.jsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default {
   },
   data() {
     return {
-      sVisible: !!this.$props.visible,
+      sVisible: !!this.$props.visible || !!this.$props.defaultVisible,
```
Comment (tangjinzhou) [inline components/tooltip/Tooltip.jsx]: 这些应该是1.4.0分支后续改动的部分

File: components/result/index.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+// ExceptionImageMap keys
+const ExceptionStatus = Object.keys(ExceptionMap);
+
+export const ResultProps = {
+  prefixCls: PropTypes.string,
+  icon: PropTypes.string,
+  status: PropTypes.oneOf(['success', 'error', 'info', 'warning', '404', '403', '500']).def('info'),
+  title: PropTypes.string,
+  subTitle: PropTypes.string,
```
Comment (tangjinzhou) [inline components/result/index.jsx]: support slot and vnode
直接用 any 吧

File: components/result/index.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  inject: {
+    configProvider: { default: () => ConfigConsumerProps },
+  },
+  methods: {
+    renderIcon(prefixCls, { status, icon }) {
+      if (ExceptionStatus.includes(status)) {
+        const svgImg = ExceptionMap[status];
+        return (
+          <div class={`${prefixCls}-icon ${prefixCls}-image`}>
+            <img alt={status} src={svgImg} />
```
Comment (tangjinzhou) [inline components/result/index.jsx]: 使用 react 版本的 函数式组件的形式 ？

File: components/result/index.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    const title = getComponentFromProp(this, 'title') || null;
+    const subTitle = getComponentFromProp(this, 'subTitle') || null;
+    const icon = getComponentFromProp(this, 'icon') || null;
+
+    return (
+      <div class={`${prefixCls} ${prefixCls}-${status}`}>
+        {this.renderIcon(prefixCls, { status, icon })}
+        <div class={`${prefixCls}-title`}>{title}</div>
+        <div class={`${prefixCls}-subtitle`}>{subTitle}</div>
```
Comment (tangjinzhou) [inline components/result/index.jsx]: {subTitle && <div class={`${prefixCls}-subtitle`}>{subTitle}</div>}

File: components/result/index.jsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        {this.$slots.default && <div class={`${prefixCls}-content`}>{this.$slots.default}</div>}
+        {this.renderExtra(prefixCls, this.$slots.extra)}
```
Comment (tangjinzhou) [inline components/result/index.jsx]: this.renderExtra(prefixCls, getComponentFromProp(this, 'extra'))

File: components/result/index.jsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      prefixCls: customizePrefixCls,
+      status,
+      ...restProps
+    } = getOptionProps(this);
+    const getPrefixCls = this.configProvider.getPrefixCls;
+    const prefixCls = getPrefixCls('result', customizePrefixCls);
+
+    const title = getComponentFromProp(this, 'title') || null;
+    const subTitle = getComponentFromProp(this, 'subTitle') || null;
+    const icon = getComponentFromProp(this, 'icon') || null;
```
Comment (tangjinzhou) [inline components/result/index.jsx]: 不需要 || null

File: components/result/index.jsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    );
+  },
+};
+
+/* istanbul ignore next */
+Result.install = function(Vue) {
+  Vue.use(Base);
+  Vue.component(Result.name, Result);
+};
+export default Result;
```
Comment (tangjinzhou) [inline components/result/index.jsx]: Result.PRESENTED_IMAGE_403 = ExceptionMap[403];
Result.PRESENTED_IMAGE_404 = ExceptionMap[404];
Result.PRESENTED_IMAGE_500 = ExceptionMap[500];

可以把这个加上，用户可以直接使用这几个图标

File: components/tooltip/Tooltip.jsx (hunk 2 above)
Comment (tangjinzhou) [reply components/tooltip/Tooltip.jsx]: 后面好几个都是，我合并下master分支吧

Comment (sendya) [reply components/tooltip/Tooltip.jsx]: 我从 1.4.0 切出来之后就是往我自己的 fork 上推送更新时，这个 commit 不知道为什么，就把你上一条的 commit 类似于 rebase 状态的带在这个 commit 上了。 我在看看

File: components/result/index.jsx (hunk 8 above)
Comment (sendya) [reply components/result/index.jsx]: 好的。

**Pull request closed without merging**